onyo/lib/onyo.py

```diff
@@ -26,12 +27,32 @@ class OnyoProtectedPathError(Exception):
 class Repo:
     """
     """
-    def __init__(self, path: Union[Path, str], init: bool = False) -> None:
+    def __init__(self, path: Union[Path, str], find_root: bool = False,
+                 init: bool = False) -> None:
+        """
+        Instantiates a `Repo` object with `path` as the root directory.
+
+        If `find_root=True` searches the root of a repository from `path`.
+
+        If `init=True`, the `path` will be initialized as a git repo and a
+        `.onyo/` directory will be created.
+        Otherwise the validity of the onyo repository is verified.
+        """
+        path = Path(path)
+
+        if find_root:
+            path = self._find_root(path)
+        else:
+            path = Path(path).resolve()
```

Comment:
`path` is already ensured above to be a `Path` object.

```suggestion
            path = path.resolve()
```